src/repositories/extendedRepo.spec.ts

```diff
@@ -0,0 +1,203 @@
+import { Column, Connection, ConnectionOptions, createConnection, Entity, PrimaryColumn } from "typeorm";
+import { BaseTypeORMRepo } from ".";
+import { RepoInterface } from "../interfaces";
+
+type TestResource = {
+    id: string;
+    name: string;
+    description: string;
+}
+
+@Entity()
+export class TestResourceEntity {
+    @PrimaryColumn() id: string;
+    @Column() name: string;
+    @Column() description: string;
+}
+
+describe('ExtendedTypeORMRepo', ()=>{
+    class ExtendedTypeORMRepo extends BaseTypeORMRepo<TestResource>{
+        constructor(connection: Connection, resourceName: string){
+            super(connection, resourceName)
+        }
+
+        create(data: TestResource): Promise<TestResource | Error> {
```

Comment:
we need to write some stuff here so that it is extended in real sense. use the example i mentioned in the morning.. like adding the validation logic to verify if the id property is a uuid or not. it can then invoke the parent class's create method